tests/integration/test_scheduler_query/test.py

```diff
@@ -194,3 +202,37 @@ def ensure_workload_concurrency(workload, limit: int) -> None:
     production.stop()
     development.stop()
     admin.stop()
+
+
+def test_max_waiting_queries_reached() -> None:
+    node.query(
+        f"""
+        create resource query (query);
+        create workload all settings max_concurrent_queries=1, max_waiting_queries=1;
+        """
+    )
+
+    pool_all = QueryPool(6, "all")
+
+    pool_all.start()
+    ensure_total_concurrency(1)
+    ensure_workload_concurrency("all", 1)
+    pool_all.stop()
+    assert "Queue limit has been reached" in pool_all.last_error
+
+
+def test_under_max_waiting_queries_limit() -> None:
+    node.query(
+        f"""
+        create resource query (query);
+        create workload all settings max_concurrent_queries=1, max_waiting_queries=6;
+        """
+    )
+
+    pool_all = QueryPool(6, "all")
```

Comment:
I think it should pass even with 7 queries: 1 running and 6 waiting
```suggestion
    pool_all = QueryPool(7, "all")
```